src/weapon/w_044.c

```diff
@@ -110,7 +110,229 @@ static s32 g_HandId = HAND_ID;
 
 #include "shared.h"
 
-INCLUDE_ASM("weapon/nonmatchings/w_044", EntityWeaponAttack);
+extern s32 D_80097420;
+extern s32 D_138000_8017A260[];
+
+void EntityWeaponAttack(Entity* self) {
+    bool doLastblock;
+    Primitive* prim;
+    s16 selfY;
+    s16 base;
+    s16 selfY_bottom;
+    s16 offset;
+    u16 vCoord;
+    s16 sine;
+    s16 angle;
+    s32 i;
+    s16 lastBlockYShift;
+    u16 clut;
+
+    clut = 0x110;
+    vCoord = 0;
+    if (g_HandId != 0) {
+        clut += 0x18;
+        vCoord += 0x80;
+    }
+
+    doLastblock = false;
+    switch (self->step) {
+    case 0:
+        self->primIndex = g_api.AllocPrimitives(
+            PRIM_GT4, 17); // bug? loops iterate over 16 prims
```

Comment:
Not a bug, it iterate over 16 primitives yes. But then after the for loop it operates on the last prim